arbnode/batch_poster.go

```diff
@@ -1126,7 +1126,7 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 		}
 		var use4844 bool
 		config := b.config()
-		if config.Post4844Blobs && b.dapWriter == nil && latestHeader.ExcessBlobGas != nil && latestHeader.BlobGasUsed != nil {
+		if config.Post4844Blobs && b.dapWriter == nil && latestHeader.ExcessBlobGas != nil && *latestHeader.ExcessBlobGas != 0 && latestHeader.BlobGasUsed != nil && *latestHeader.BlobGasUsed != 0 {
```

Comment:
are the last two conditions right?
I think we might post blobs even if BlobGasUsed == 0 (possibly even if it's nil? IDK)

Reply:
I don't really remember the reason but without this check batch poster was failing for some reason.

Adding @PlasmaPower to confirm if he remembers what was the reason.

Reply:
No, I don't know the reason for this, but this change looks a bit scary. I'd try reverting it and see what fails, but I suspect the fix should be elsewhere.